lib/vehicle_factory.rb

```diff
@@ -0,0 +1,344 @@
+require './lib/vehicle'
+require './lib/vehicle_factory'
+require './lib/dmv_data_service'
+require 'faraday'
+require 'json'
+require 'pry'
+
+class VehicleFactory
+
+    def initialize
+  end
+# Most proud of this create_vehicles method:
+  def create_vehicles(source)
+    if source.find {|car| car[:state_of_residence] == "WA"}
+      nc_arr = []
+      source.each do |car|
+        new_car = {
+          vin: car[:vin_1_10],
+          year: car[:model_year],
+          make: car[:make],
+          model: car[:model],
+          engine: :ev}
+        nc_arr << Vehicle.new(new_car)
+      end
+      nc_arr
+    elsif source.find {|car| car[:state] == "NY"}
+      nc_arr = []
+      source.each do |car|
+        new_car = {
+          vin: car[:vin],
+          year: car[:model_year],
+          make: car[:make],
+          body_type: car[:body_type],
+          engine: :ev}
+        nc_arr << Vehicle.new(new_car)
+      end
+      nc_arr
+    else
+      "N/A"
+    end
+  end
+
+  def find_most_common(source) # I plan to refactor and correct this!
+    by_make = []
+    by_model = []
+    by_make_and_model = []
+    source.each do |car|
+      car_data = {
+        make: car[:make],
+        model: car[:model],
+        make_and_model: car[:make] + ', ' + car[:model],
+      }
+      by_make << car_data[:make]
+      by_model << car_data[:model]
+      by_make_and_model << car_data[:make_and_model]
+    end
+
+    tesla_model_s = []
+    nissan_leaf = []
+    toyota_prius_plug = []
+    volvo_xc90 = []
+    tesla_x = []
+    chevy_volt = []
+    fiat_500 = []
+    tesla_3 = []
+    ford_mustang_mach_e = []
+    tesla_model_y = []
+    bmw_i3 = []
+    audi_e_tron = []
+    smart_fortwo_electric = []
+    audi_q5_e = []
+    bmw_330e = []
+    kia_soul = []
+    bmw_i8 = []
+    ford_escape = []
+    mercedes_benz_glc_class = []
+    mercedes_benz_eqb_class = []
+    chevy_bolt_ev = []
+    honda_clarity = []
+    volkswagen_id4 = []
+    kia_niro = []
+    porsche_taycan = []
+    volkswagen_egolf = []
+    toyota_prius_prime = []
+    jeep_wrangler = []
+    chevy_bolt_euv = []
+    ford_fusion = []
+    volvo_xc60 = []
+    volvo_xc40 = []
+    hyundai_ioniq5 = []
+    ford_transit = []
+    audi_a3 = []
+    hyundai_ioniq = []
+    bmw_x5 = []
+    porsche_cayenne = []
```

Comment:
I don't think you need any of this with the approach you're taking. Rather, I'd suggest doing something like this in the map below: 

```ruby 
 cars_by_model = Hash.new()
 by_make_and_model.map do |car|
   cars_by_model[car]  += 1
 end 
...
```

this would create keys for the car's make/model, and increment one every time that make/model came through.

Reply:
You could apply this idea to a lot of the code you have for Iteration 4.